Modules/_blake2/impl/blake2-impl.h

```diff
@@ -140,6 +140,9 @@ static inline void secure_zero_memory(void *v, size_t n)
 {
 #if defined(_WIN32) || defined(WIN32)
   SecureZeroMemory(v, n);
+#elif defined(__hpux)
+  static void *(*const volatile memset_v)(void *, int, size_t) = &memset;
+  memset_v(v, 0, n);
```

Comment:
How is it different from just `memset(v, 0, n)`?

This code looks different from https://github.com/BLAKE2/libb2/commit/b4b241a34824b51956a7866606329a065d397525

Reply:
there was a recent change to fix HP-UX build which is backported in [this PR](https://github.com/BLAKE2/libb2/blob/master/src/blake2-impl.h#L144).
There is a subtle difference, this prevents the compiler to optimise away the buffer clearing, the volatile function pointer plays its role here.

Reply:
I got [reply in upstream](https://github.com/BLAKE2/libb2/pull/26#discussion_r288364105) too.

While I like comment here, it's up to upstream project.  We should copy code as-is to reduce maintenance cost.